crates/re_types/src/lib.rs

```diff
@@ -172,6 +172,7 @@
 //!
 
 #![warn(missing_docs)] // Let's keep the this crate well-documented!
+#![deny(clippy::unwrap_used)] // See #3408, remove once we remove all the unwrap's
```

Comment:
Why is this needed? In other crates that aren't done yet you correctly opt-out of the warning, but here the warning is opted in to error. Ci won't pass where we got clippy warnings, so this seems redundant in a way

Reply:
The idea is to mark this crate as "unwrap safe" already. All other (unfinished) crates should warn and the ones where unsafe is safe (😄...tests, benchmarks, builders, etc) should explicitly opt-out and allow unwrap().

Reply:
I think there's two issues with this:
* unwrap_used being only a warning is nice because it means that during development I can take shortcuts, but upon PR submission I have to clean this all up (ci won't let anything through with warnings)
* this means we have both opt-out and opt-in to deal with at the same time